tests/integration/chroot.py

```diff
@@ -0,0 +1,298 @@
+        if chroot_epoch_bool:
+            logger.warning("'garden-epoch' already present. This will be set to 'dev', now.")
+        try:
+            with open(chroot_epoch, 'w') as f:
+                f.write("dev")
+            logger.info("Wrote garden-epoch to chroot in {tmp_dir}".format(
+              tmp_dir=tmp_dir))
+        except IOError:
+            logger.error("Could not write garden-epoch to chroot in {tmp_dir}".format(
+              tmp_dir=tmp_dir))
+        # Generate SSH hostkeys for chroot
+        # (this way we do not need to execute this inside the 'chroot'
+        #  environment and still support amd64 and arm64 architectures)
+        chroot_ssh_dir = tmp_dir + "/etc/ssh/"
+        cmd_ssh_keys = []
+        cmd_ssh_keys.append('ssh-keygen -q -N "" -t dsa -f {ssh_dir}/ssh_host_dsa_key'.format(
+          ssh_dir=chroot_ssh_dir))
+        logger.info(cmd_ssh_keys)
+        cmd_ssh_keys.append('ssh-keygen -q -N "" -t rsa -b 4096 -f {ssh_dir}/ssh_host_rsa_key'.format(
+          ssh_dir=chroot_ssh_dir))
+        cmd_ssh_keys.append('ssh-keygen -q -N "" -t ecdsa -f {ssh_dir}/ssh_host_ecdsa_key'.format(
+          ssh_dir=chroot_ssh_dir))
+        cmd_ssh_keys.append('ssh-keygen -q -N "" -t ed25519 -f {ssh_dir}/ssh_host_ed25519_key'.format(
+          ssh_dir=chroot_ssh_dir))
+        for i in cmd_ssh_keys:
+            logger.info("Running command: {cmd}".format(
+              cmd=i))
+            p = subprocess.run([i], shell=True, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+            rc = p.returncode
+            if rc != 0:
+                logger.error("Command failed.")
+            else:
+                logger.info("Command sucessfully executed.")
+        # Generate chroot sshd directory for run/pid
+        chroot_sshd_run_dir = tmp_dir + "/run/sshd"
+        try:
+            os.mkdir(tmp_dir + "/run/sshd")
+            logger.info("Created directory: {dir}".format(
+              dir=chroot_sshd_run_dir))
+        except OSError:
+            logger.error("Could not create directory: {dir}".format(
+              dir=chroot_sshd_run_dir))
+        # Copy dedicated sshd config to chroot
+        sshd_config_src_file = "integration/misc/sshd_config_integration_tests"
+        chroot_sshd_cfg_dir = tmp_dir + "/etc/ssh/"
+        try:
+            shutil.copyfile(sshd_config_src_file, chroot_sshd_cfg_dir+"sshd_config_integration_tests")
+            logger.info("Copied sshd_config_integration_tests to: {dir}".format(
+              dir=chroot_sshd_cfg_dir))
+        except OSError:
+            logger.error("Could not copy sshd_config_integration_tests to {dir}".format(
+              dir=chroot_sshd_cfg_dir))
+        # Copy ssh / authorized_keys file for root user
+        chroot_root_dir = tmp_dir + "/root/"
+        chroot_ssh_authorized_keys = chroot_root_dir + ".ssh/authorized_keys"
+        ssh_authorized_keys = "/tmp/authorized_keys"
+        self._create_dir(chroot_root_dir+".ssh", 0o600)
```

Comment:
I would use the following command here:
```
access = 0o600
os.mkdir(path, access)
```
Then, there is no need for this helper function